server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java

```diff
@@ -70,82 +75,78 @@ public boolean intersects(Extent extent) throws IOException {
     }
 
     boolean containsBottomLeft(Extent extent) throws IOException {
-        resetInputPosition();
-
-        Optional<Boolean> extentCheck = checkExtent(new Extent(input), extent);
+        Optional<Boolean> extentCheck = checkExtent(getExtent(), extent);
         if (extentCheck.isPresent()) {
             return extentCheck.get();
         }
 
-        return containsBottomLeft(readRoot(input.position()), extent);
+        resetToRootEdge();
+        if (input.readBoolean()) { /* has edges */
+            return containsBottomLeft(input.position(), extent);
+        }
+        return false;
     }
 
     boolean containsFully(Extent extent) throws IOException {
-        resetInputPosition();
-        input.position(input.position() + Extent.WRITEABLE_SIZE_IN_BYTES); // skip extent
-        return containsFully(readRoot(input.position()), extent);
+        getExtent();
```

Comment:
do we need this?

Reply:
this will be unnecessary after I merge in the rasterization changes that reduces the surfaces area of methods that require the Extent.

The reason I chose to lazily load it in this PR is because the geo-centroid aggregation does not care about the extent